src/gt4py/next/program_processors/runners/dace_fieldview/transformations/map_serial_fusion.py

```diff
@@ -0,0 +1,475 @@
+                    shape=new_inter_shape,
+                    dtype=inter_desc.dtype,
+                    find_new_name=True,
+                )
+            new_inter_node: nodes.AccessNode = state.add_access(new_inter_name)
+
+            # New we will reroute the output Memlet, thus it will no longer going
+            #  through the Map exit but through the newly created intermediate.
+            #  we will delete the previous edge later.
+            pre_exit_memlet: dace.Memlet = pre_exit_edge.data
+            new_pre_exit_memlet = copy.deepcopy(pre_exit_memlet)
+
+            # We might operate on a different array, but the check below, ensures
+            #  that we do not change the direction of the Memlet.
+            assert pre_exit_memlet.data == inter_name
+            new_pre_exit_memlet.data = new_inter_name
+
+            # Now we have to fix the subset of the Memlet.
+            #  Before the subset of the Memlet dependent on the Map variables,
+            #  however, this is no longer the case, as we removed them. This change
+            #  has to be reflected in the Memlet.
+            #  NOTE: Assert above ensures that the bellow is correct.
+            new_pre_exit_memlet.replace(memlet_repl)
+            if is_scalar:
+                new_pre_exit_memlet.subset = "0"
+                new_pre_exit_memlet.other_subset = None
+            else:
+                new_pre_exit_memlet.subset.pop(squeezed_dims)
+
+            # Now we create the new edge between the producer and the new output
+            #  (the new intermediate node). We will remove the old edge further down.
+            new_pre_exit_edge = state.add_edge(
+                pre_exit_edge.src,
+                pre_exit_edge.src_conn,
+                new_inter_node,
+                None,
+                new_pre_exit_memlet,
+            )
+
+            # We just have handled the last Memlet, but we must actually handle the
+            #  whole producer side, i.e. the scope of the top Map.
+            for producer_tree in state.memlet_tree(new_pre_exit_edge).traverse_children():
+                producer_edge = producer_tree.edge
+
+                # Ensure the correctness of the rerouting below.
+                # TODO(phimuell): Improve the code below to remove the check.
+                assert producer_edge.data.data == inter_name
+
+                # Will not change the direction, because of test above!
+                producer_edge.data.data = new_inter_name
+                producer_edge.data.replace(memlet_repl)
+                if is_scalar:
+                    producer_edge.data.dst_subset = "0"
+                else:
+                    if producer_edge.data.dst_subset is not None:
+                        producer_edge.data.dst_subset.pop(squeezed_dims)
+
+            # Now after we have handled the input of the new intermediate node,
+            #  we must handle its output. For this we have to "inject" the temporary
```

Comment:
temporary -> intermediate (just for consistent terminology, assuming I've understood it right)

Reply:
You were right.
I have adapted the wording.